molgenis-python/src/main/resources/templates/molgenis.py.ftl

```diff
@@ -158,3 +175,18 @@ class Session():
         z = x.copy()
         z.update(y)
         return z
+
+    def _raise_for_status(self, response):
+        error = None
+        try:
+            if 'errors' in response.json() and 'message' in response.json()['errors'][0]:
+                error = (response.json()['errors'][0]['message'])
+        except json.decoder.JSONDecodeError:
+            pass
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            if error:
+                raise requests.exceptions.HTTPError(str(e)+'\n'+error)
```

Comment:
If you catch the HttpError e then the response will be in e.response and the caller gets complete control over what to do with it. So for instance:
```
>>> try:
...   c.login('a','b')
... except requests.exceptions.HTTPError as e:
...   print(e.response.json())

{u'errors': [{u'message': u"org.springframework.security.core.userdetails.UsernameNotFoundException: unknown user 'a'"}]}
```